### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by seriva**
Initial implementation:

- Tested on AKS, EKS basic settings with demo application on top.
- For EKS used this [pull](https://github.com/epiphany-platform/m-aws-kubernetes-service/pull/30) and set node size to m.medium.

### CODE REVIEW

File: resources/templates.mk [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  namespace: $(M_FIRST_WATCHED_NAMESPACES)
+  name: $(M_CONTROLPLANE_NAME)
+spec:
+  profile: $(M_CONTROLPLANE_PROFILE)
+  values:
+    global:
+       istioNamespace: $(M_ISTIO_NAMESPACE)
```
Comment (rpudlowski93) [inline resources/templates.mk:49]: I think, we need to add here hub parameter in order to make possible using external/provided custom hub with images also for istio (control plane, proxy etc..). In istioctl, flag : --hub is related only to istio operator.

Comment (seriva) [reply resources/templates.mk:49]: Yes I will be adding that today:)

Comment (seriva) [reply resources/templates.mk:49]: Fixed.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,194 @@
...
+- [Azure Kubernetes Servicee](https://github.com/epiphany-platform/m-azure-kubernetes-service#run-module)
+
+**Note : On AWS EKS make sure to have at least ```t.medium``` as node size**
+
+At this stage you should have already /tmp/shared directory with your ssh-keys, module data from the AKS/EKS deployments and the kubeconf file.
+
+* Initialize the Istio module:
+
+  ```shell
+  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/Istio:latest init
```
Comment (rpudlowski93) [inline README.md:43]: I think that repository name must be lowercase

**Comment by rpudlowski93** [comment]
Great job! I tested it on AKS and everything works fine :)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,194 @@
...
+  cd m-aws-basic-infrastructure/
+  docker build --tag epiphanyplatform/istio:latest .
+  ```
+
+## Run module
+
+This module requires that you to already have a deployed EKS or AKS cluster using the following modules/instructions:
+
+- [AWS Kubernetes Service](https://github.com/epiphany-platform/m-aws-kubernetes-service#run-module)
+- [Azure Kubernetes Servicee](https://github.com/epiphany-platform/m-azure-kubernetes-service#run-module)
```
Comment (atsikham) [inline README.md:34]:
```suggestion
- [Azure Kubernetes Service](https://github.com/epiphany-platform/m-azure-kubernetes-service#run-module)
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,194 @@
...
+In main directory run:
+
+  ```shell
+  make build
+  ```
+
+or directly using Docker:
+
+  ```shell
+  cd m-aws-basic-infrastructure/
```
Comment (atsikham) [inline README.md:25]:
```suggestion
  cd m-istio/
```

**Review by atsikham — CHANGES_REQUESTED** [review]
I'm verifying the default installation in EKS, will report status later.

File: README.md (hunk 2 above)
Comment (atsikham) [reply README.md:43]: It at least in lowercase on the build stage

**Review by atsikham — CHANGES_REQUESTED** [review]

**Comment by atsikham** [comment]
Do we need to exclude some files from Docker build context? There is no `.dockerignore` now.

**Comment by atsikham** [comment]
After a [fix](https://github.com/epiphany-platform/m-aws-kubernetes-service/pull/35) in AWS KS module I was able to rebuild EKS image and then deployed a default configuration on EKS.
image

File: README.md (hunk 2 above)
Comment (seriva) [reply README.md]: Fixed:)

**Comment by seriva** [comment]
> Do we need to exclude some files from Docker build context? There is no `.dockerignore` now.

Ill check it.

**Comment by seriva** [comment]
> After a [fix](https://github.com/epiphany-platform/m-aws-kubernetes-service/pull/35) in AWS KS module I was able to rebuild EKS image and then deployed a default configuration on EKS.
> image

Jup I ran into the same issue yesterday and made the same fix:D

**Comment by seriva** [comment]
> > Do we need to exclude some files from Docker build context? There is no `.dockerignore` now.
> 
> Ill check it.

Added dockerignore.

File: workdir/Makefile [hunk 5]
```diff
@@ -0,0 +1,220 @@
+include $(M_RESOURCES)/templates.mk
+include $(M_RESOURCES)/consts.mk
+include $(M_RESOURCES)/defaults.mk
+
+export AWS_ACCESS_KEY_ID=$(M_AWS_ACCESS_KEY)
+export AWS_SECRET_ACCESS_KEY=$(M_AWS_SECRET_KEY)
```
Comment (mkyc) [inline workdir/Makefile:6]: why do we need it?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,194 @@
...
+
+### AKS
+
+* Prepare your own variables in vars.mk file to use in the building process. Sample file (examples/basic_flow_aks/vars.mk.sample):
+
+  ```shell
+  ARM_CLIENT_ID ?= "appId field"
+  ARM_CLIENT_SECRET ?= "password field"
+  ARM_SUBSCRIPTION_ID ?= "id field"
+  ARM_TENANT_ID ?= "tenant field"
```
Comment (mkyc) [inline README.md:96]: why do I need those values in this module?

File: docs/INPUTS.adoc [hunk 7]
```diff
@@ -0,0 +1,31 @@
+== Input parameters
+
+[width="100%",cols="7%,1%,100%a,1%,100%a,50%a",options="header",]
+|===
+|Name |Type |Default value |Required |Steps |Description
+|AWS_ACCESS_KEY |string |unset |yes |plan, apply, plan-destroy, destroy |Access key id 
+(Only needed when using an EKS cluster)
+
+|AWS_SECRET_KEY |string |unset |yes |plan, apply, plan-destroy, destroy |Access key secret 
+(Only needed when using an EKS cluster)
```
Comment (mkyc) [inline docs/INPUTS.adoc:10]: why do I need those values in this module?

**Review by mkyc — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -1,2 +1,194 @@
...
+  ```shell
+  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/istio:latest init
+  ```
+
+  This commad will create configuration file of AwsKS module in /tmp/shared/istio/istio-config.yml. You can investigate what is stored in that file.
+  Available parameters are listed in the [inputs](docs/INPUTS.adoc) document.
+
+* Plan and apply Istio module:
+
+**Note : The ```M_AWS_ACCESS_KEY``` and ```M_AWS_SECRET_KEY``` environment variables are only needed for EKS**
```
[the remark this answers is not in the document]
Comment (seriva) [reply README.md:51]: In order to use the kubeconf generated by EKS for both istioctl and kubectl you need AWS cli which in turn needs the credentials:




Tested both from module and normal install.

File: workdir/Makefile (hunk 5 above)
Comment (seriva) [reply workdir/Makefile:6]: See comment here: https://github.com/epiphany-platform/m-istio/pull/1#discussion_r552094827

File: README.md (hunk 6 above)
Comment (seriva) [reply README.md:96]: See comment here: https://github.com/epiphany-platform/m-istio/pull/1#discussion_r552094827

File: docs/INPUTS.adoc (hunk 7 above)
Comment (seriva) [reply docs/INPUTS.adoc:10]: See comment here: See comment here: https://github.com/epiphany-platform/m-istio/pull/1#discussion_r552094827

**Comment by seriva** [comment]
> > Do we need to exclude some files from Docker build context? There is no `.dockerignore` now.
> 
> Ill check it.

Added `.dockerignore`

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (rpudlowski93) [reply README.md:51]: Indeed I also remember that in contrary to Azure/AKS, You need to have configured aws using credential in order to use kubectl with kubeconf generatred by EKS.  I think that there is a way to run kubectl without passing the credentials but the kubeconfig must be updated as it is mentioned here [link](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html). For our usage I suppose that this way which Luuk implemented is enough.

**Review by rpudlowski93 — APPROVED** [review]
LGTM 👍

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply README.md:51]: I added issue #2 to reconsider it in future.

File: README.md (hunk 6 above)
Comment (mkyc) [reply README.md:96]: Wait a sec, those are for AKS not EKS.

Comment (seriva) [reply README.md:96]: Ahh I see, it's running of examples which need the deploy BI and AKS first.

Comment (mkyc) [reply README.md:96]: Ahhh .... ok, I didn't notice that it's about examples. I believe we will need to make documentation more clear about it in future, but for now it's OK.

**Review by mkyc — APPROVED** [review]
LGTM

**Review by atsikham — APPROVED** [review]

**Pull request merged**